mantidproject/mantid
# Plot selection in muon GUI - prototype

**Description of work.**
This changes the way the muon GUI will display plots on the data pane. Users now want to load 10's of runs (up to 100) at once, which made the plots difficult to read. These changes will only plot one line by default and allow the user to select the lines they want to see.
 
**To test:**


Open muon analysis
Load a single data (e.g. MUSR62260)
Only one line should be plotted
Click the select data  button on the plotting window
A pop up will appear
If you click a row, it will plot the corresponding data
Check it works with holding shift (selects everything between clicks) and ctl (select these)

Select a couple of lines
Load some more data, but include the data you are looking at (e.g. MUSR 62260-1)
The plot should not change
Check that the plot selection has updated

Load some data that does not include any of the data you have (e.g. MUSR 62250-1)
Only one line will be plotted

Go to the fitting tab
Add a fit function
Go to sequential fitting
Check it looks ok and you can edit it
Do a sequential fit - check it looks ok.


Fixes #33052. 







---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
